pkg/pilot/elasticsearch/v5/controller.go

```diff
@@ -39,3 +38,8 @@ func (p *Pilot) syncFunc(pilot *v1alpha1.Pilot) error {
 	}
 	return nil
 }
+
+func (p *Pilot) leaderElectedSyncFunc(pilot *v1alpha1.Pilot) error {
+	glog.V(4).Infof("ElasticsearchController: leader elected sync of pilot %q", pilot.Name)
+	return nil
+}
```

Comment:
What operations might be performed here?
https://github.com/jetstack/navigator/issues/97 talks about `excludeShards`, but that seems like something that would be performed by the Pilot that had been selected to leave the cluster  by the ES controller.

Reply:
So in the ES pilot case, excludeShards will be handled here. I've got some more work locally to actually do this. This needs to be leader elected as it requires updating the ES API with the new list of nodes that should be excluded. We *could* do it in each individual Pilot and just hope there's no races, but this way is far more concrete.